src/main/java/core/basesyntax/FIgureSupplier.java

```diff
@@ -0,0 +1,41 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FIgureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureType = random.nextInt(5);
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureType) {
+            case 0:
+                return new Square(color,
+                        random.nextDouble() * 10 + 1);
```

Comment:
Instead of passing expressions directly into the constructor, create separate variables for the size of the 'Square' and pass them to the constructor for better code readability.